### PR TITLE
CLN: use fused type for lib.has_infs

### PR DESCRIPTION
**Pull request by mzeitlin11**

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -519,36 +520,22 @@ def get_reverse_indexer(const intp_t[:] indexer, Py_ssize_t length) -> ndarray:
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def has_infs_f4(const float32_t[:] arr) -> bool:
+# Can add const once https://github.com/cython/cython/issues/1772 resolved
+def has_infs(floating[:] arr) -> bool:
     cdef:
         Py_ssize_t i, n = len(arr)
         float32_t inf, neginf, val
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:527]: should this be `floating inf, neginf, val`?

Comment (mzeitlin11) [reply pandas/_libs/lib.pyx:527]: Thanks, definitely!

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**